spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/AnalyzeTableSparkAction.java

```diff
@@ -0,0 +1,159 @@
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.spark.JobGroupInfo;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Computes the statistic of the given columns and stores it as Puffin files. */
+public class AnalyzeTableSparkAction extends BaseSparkAction<AnalyzeTableSparkAction>
+    implements AnalyzeTable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnalyzeTableSparkAction.class);
+
+  private final Table table;
+  private Set<String> columns = ImmutableSet.of();
+  private Set<String> types = StandardBlobTypes.blobTypes();
+  private Long snapshotId;
+
+  AnalyzeTableSparkAction(SparkSession spark, Table table) {
+    super(spark);
+    this.table = table;
+  }
+
+  @Override
+  protected AnalyzeTableSparkAction self() {
+    return this;
+  }
+
+  @Override
+  public Result execute() {
+    if (snapshotId == null) {
+      snapshotId = table.currentSnapshot().snapshotId();
+    }
+    String desc = String.format("Analyzing table %s for snapshot id %s", table.name(), snapshotId);
+    JobGroupInfo info = newJobGroupInfo("ANALYZE-TABLE", desc);
+    return withJobGroupInfo(info, this::doExecute);
+  }
+
+  private Result doExecute() {
+    LOG.info("Starting the analysis of {} for snapshot {}", table.name(), snapshotId);
+    List<AnalysisResult> analysisResults =
+        types.stream()
+            .map(
+                statsName -> {
+                  switch (statsName) {
+                    case StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1:
+                      return generateNDVAndCommit();
+                    default:
+                      return ImmutableAnalyzeTable.AnalysisResult.builder()
+                          .type(statsName)
+                          .addAllErrors(Lists.newArrayList("Stats type not supported"))
+                          .build();
+                  }
+                })
+            .collect(Collectors.toList());
+    return ImmutableAnalyzeTable.Result.builder().analysisResults(analysisResults).build();
+  }
+
+  private boolean analyzableTypes(Set<String> columnNames) {
+    return columnNames.stream()
+        .anyMatch(
+            columnName -> {
+              Types.NestedField field = table.schema().findField(columnName);
+              if (field == null) {
+                throw new ValidationException("No column with %s name in the table", columnName);
+              }
+              Type.TypeID type = field.type().typeId();
+              return type == Type.TypeID.INTEGER
+                  || type == Type.TypeID.LONG
+                  || type == Type.TypeID.STRING
+                  || type == Type.TypeID.DOUBLE;
```

Comment:
Why only these?

Reply:
These were the datatypes supported by the sketch lib. I think this no longer relevant with `Conversions.toByteBuffer`